prov/efa/src/rdm/rxr_env.c

```diff
@@ -281,39 +248,40 @@ void rxr_env_initialize()
 /**
  * @brief Fetch the value of the environment variable FI_EFA_USE_DEVICE_RDMA
  *
- * This function fetches the change in the value of FI_EFA_USE_DEVICE_RDMA
- * if any, after it was defined and set in rxr_env_define_use_device_rdma() 
- * 
+ * This function fetches the value of FI_EFA_USE_DEVICE_RDMA from the
+ * user's environment.
+ *
+ * The return value is not boolean.  It may be 1, 0, or -FI_ENODATA.
+ *
+ * Environments with non-boolean (1, true, yes, on) values for the variable
+ * will print to stderr and abort the program.
+ *
  * @return		0 - If FI_EFA_USE_DEVICE_RDMA is set to 0/false/no/off
- * 			1 - If FI_EFA_USE_DEVICE_RDMA is set to 1/true/yes/on
- * 			If FI_EFA_USE_DEVICE_RDMA is not set, then the value
- * 			returned from rxr_env_get_default_use_device_rdma()
- * 			will be the return value of this function.
+ *			1 - If FI_EFA_USE_DEVICE_RDMA is set to 1/true/yes/on
+ *			-FI_ENODATA - if FI_EFA_USE_DEVICE_RDMA is not set.
  */
 int rxr_env_get_use_device_rdma()
 {
-	int ret, use_device_rdma;
+	int ret;
+	int param_val;
 
-	/* Get the default value of using device's RDMA capability */
-	use_device_rdma = rxr_env_get_default_use_device_rdma();
 
 	/* Fetch the value of environment variable set by the user if any. */
-	ret = fi_param_get_bool(&efa_prov, "use_device_rdma", &use_device_rdma);
+	ret = fi_param_get_bool(&efa_prov, "use_device_rdma", &param_val);
 	if (ret == -EINVAL){
 		fprintf(stderr, "FI_EFA_USE_DEVICE_RDMA was set to an invalid value by the user."
 			" FI_EFA_USE_DEVICE_RDMA is boolean and can be set to only 0/false/no/off or"
 			" 1/true/yes/on.\n");
 		abort();
 	}
 
-	/* When the default value was to not use it, but the user sets
-	 * the value to use device RDMA in a device that does not support
-	 * RDMA, exit the run */
-	if (ret != -FI_ENODATA && use_device_rdma &&
+	/* When the user requests use device RDMA but the device does not
+	   support RDMA, exit the run. */
+	if (ret == 1 &&
```

Comment:
This does not look right. I think you want to check `param_val`. 

`ret` is the return value of the function

Reply:
Yes, thank you!  Fixed.